Clarified issues when building miri with a custom rustc (#2112)
rust-lang/miri: CONTRIBUTING.md

rustup override set stage2
```

> Important: You need to delete the miri cache (located at `~/.cache/miri`) when

Author: I do not know where the cache would be under windows.

> Important: You need to delete the miri cache (located at `~/.cache/miri`) when

Reviewer: ```suggestion
Important: You need to delete the miri cache (located at `~/.cache/miri` on Linux; the exact location is printed after the library build: "A libstd for Miri is now available in ...") when
```

> Important: You need to delete the miri cache (located at `~/.cache/miri` on Linux; the exact location is printed after the library build: "A libstd for Miri is now available in ...") when
> you change the stdlib, otherwise the old, chached version will be used.

Reviewer: ```suggestion
Important: You need to delete the Miri cache when you change the stdlib; otherwise the
old, chached version will be used. On Linux, the cache is located at `~/.cache/miri`; the exact
location is printed after the library build: "A libstd for Miri is now available in ...".
```

Note: `./x.py --stage 2 compiler/rustc` currently errors with `thread 'main'

> panicked at 'fs::read(stamp) failed with No such file or directory (os error 2)`,
> after This should not be an issue for any development with miri.

Reviewer: ```suggestion
panicked at 'fs::read(stamp) failed with No such file or directory (os error 2)`,
you can simply ignore that error; Miri will build anyway.
```
